homework-submission/week1/src/stores/todosStore.js

```diff
@@ -0,0 +1,54 @@
+import { observable, action, configure } from 'mobx';
+import todoObj from '../todo.json';
+
+configure({
+  enforceActions: true
+});
+
+class TodosStore {
+  @observable todos = todoObj;
+  @observable editing = -1;
+
+  @action checkBoxHandler = (id) => {
+    const result = this.todos.map(item => {
+      if (item.id === parseInt(id, 10)) item.done = !item.done;
+      return item;
+    });
+    this.todos = result;
+  }
+
+  @action deleteHandler = (id) => {
+    const result = this.todos.filter(item => item.id !== parseInt(id, 10));
+    this.todos = result;
+  }
+
+  @action editHandler = (id) => {
+    this.editing = id ? id : -1;
+  }
+
+  @action updateHandler = (id, newDescription) => {
+    this.editing = -1;
+    if (!(newDescription.length === 0 || /^\s*$/.test(newDescription))) {
```

Comment:
Same note [w3](https://github.com/HackYourFuture/React/pull/215#discussion_r216128567)
And try to use [return early pattern](http://blog.timoxley.com/post/47041269194/avoid-else-return-early)

    this.editing = -1;
    if (newDescription.length === 0 || /^\s*$/.test(newDescription))
      return;

    const result = this.todos.map(item => {
      if (item.id === parseInt(id, 10)) item.description = newDescription;
      return item;
    });

    this.todos = result;